File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/07 - Nested Conditional Statements Exercise/04. New House .py
flower_type = input()
count = int(input())
budget = int(input())

price = 0

if flower_type == "Roses":
    price = 5 * count
    if count > 80:
        price -= price * 0.1
elif flower_type == "Dahlias":
    price = 3.8 * count
    if count > 90:
        price -= price * 0.15
elif flower_type == "Tulips":
    price = 2.8 * count
    if count > 80:
        price -= price * 0.15
elif flower_type == "Narcissus":
    price = 3 * count
    if count < 120:
        price += price * 0.15
elif flower_type == "Gladiolus":
    price = 2.5 * count
    if count < 80:
        price += price * 0.2

if budget >= price:
    print(f"Hey, you have a great garden with {count} {flower_type} and {(budget - price):.2f} leva left.")
else:
    print(f"Not enough money, you need {(price - budget):.2f} leva more.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/10. УПРАЖНЕНИЕ ПО-СЛОЖНИ ПРОВЕРКИ/04. New House.js
function newHouse(...input) {
    let flower = input.shift();
    let count = Number(input.shift());
    let budget = Number(input.shift());

    let price = 0;

    switch(flower) {
        case "Roses":
            if (count <= 80) {
                price = 5 * count;
            } else {
                price = (5 * count) * 0.9;
            }
            break;
        case "Dahlias":
            if (count <= 90) {
                price = 3.80 * count;
            } else {
                price = (3.80 * count) * 0.85;
            }
            break;
        case "Tulips":
            if (count <= 80) {
                price = 2.80 * count;
            } else {
                price = (2.80 * count) * 0.85;
            }
            break;
        case "Narcissus":
            if (count < 120) {
                price = (3 * count) * 1.15;
            } else {
                price = 3 * count;
            }
            break;
        case "Gladiolus":
            if (count < 80) {
                price = (2.50 * count) * 1.2;
            } else {
                price = 2.50 * count;
            }
    }

    if (price <= budget) {
        console.log(`Hey, you have a great garden with ${count} ${flower} and ${(budget -price).toFixed(2)} leva left.`);
    } else {
        console.log(`Not enough money, you need ${(price - budget).toFixed(2)} leva more.`);
    }
}

newHouse(['Roses', '55', '250']);  // Expected output:
// Not enough money, you need 25.00 leva more.

newHouse(['Tulips', '88', '260']);  // Expected output:
// Hey, you have a great garden with 88 Tulips and 50.56 leva left.

newHouse(['Narcissus', '119', '360']);  // Expected output:
// Not enough money, you need 50.55 leva more.
